packages/frontend/src/redux/slices/tokenFiatValues/index.js

```diff
@@ -4,14 +4,38 @@ import { createSelector } from 'reselect';
 
 import { ACCOUNT_HELPER_URL } from '../../../config';
 import sendJson from '../../../tmp_fetch_send_json';
+import { fetchTokenPrices, fetchTokenWhiteList } from '../../../utils/ref-finance';
 import handleAsyncThunkStatus from '../../reducerStatus/handleAsyncThunkStatus';
 import initialStatusState from '../../reducerStatus/initialState/initialStatusState';
 
 const SLICE_NAME = 'tokenFiatValues';
 
 const fetchTokenFiatValues = createAsyncThunk(
     `${SLICE_NAME}/fetchTokenFiatValues`,
-    () => sendJson('GET', ACCOUNT_HELPER_URL + '/fiat')
+    async () => {
+        const nearFiatValue = await sendJson('GET', ACCOUNT_HELPER_URL + '/fiat');
+        // TODO: Rewrite this function call on by demand or even create separate action.
+
+        const tokenPrices = await fetchTokenPrices();
+        const last_updated_at = Date.now() / 1000; 
+
+        const tokenFiatValues = Object.keys(tokenPrices).reduce((acc, curr) => {
```

Comment:
**Not a blocker for this PR**, but this can just be replaced with `_. mapValues` for clarity & brevity here. Using `Object.keys()` here makes our reducer function impure which is undesirable; if we're going to reference variables in the parent function scope, then doing a for loop or using forEach would be more idiomatic.